tests/syntax/test_operators.py

```diff
@@ -317,12 +537,27 @@ def test_visible():
 
 def test_not_visible():
     scenario = compileScenic("""
-        ego = Object at 100 @ 200, facing -45 deg,
+        ego = new Object at 100 @ 200, facing -45 deg,
                      with visibleDistance 30, with viewAngle 90 deg
         reg = RectangularRegion(100@200, 0, 10, 10)
-        param p = Point in not visible reg
+        param p = new Point in not visible reg
     """)
     ps = [sampleParamP(scenario, maxIterations=100) for i in range(50)]
     assert all(p.x <= 100 or p.y <= 200 for p in ps)
     assert any(p.x > 100 for p in ps)
     assert any(p.y > 200 for p in ps)
+
+# visible from
+
+def test_visible_from():
+    scenario = compileScenic("""
+        ego = new Object at 100 @ 200, facing -45 deg,
+                     with visibleDistance 10, with viewAngle 90 deg
+        reg = RectangularRegion(100@205, 0, 10, 20)
+        param p = new Point in reg visible from ego
+    """)
+    for i in range(30):
+        p = sampleParamP(scenario, maxIterations=100)
```

Comment:
Shouldn't this scenario never need more than one iteration?

Reply:
As we discussed, the rejection sampling the visible region sometimes raises an exception. I'll lower all these to 10 though.